### PR TITLE
Contributors Page

### PR DESCRIPTION
**Pull request by ryanchou-dev**
ready for review :)

### CODE REVIEW

File: src/components/Index/ContributorsSection.tsx [hunk 1]
```diff
@@ -4,112 +4,85 @@ import {
...
-                  className="text-blue-600 dark:text-blue-400"
-                >
-                  {author.name}
-                </a>{' '}
-                (since {author.sinceDate})
-              </li>
-            ))}
-            <li className="italic">
-              This could be you!{' '}
+    <div>
```
Comment (nathangong) [inline src/components/Index/ContributorsSection.tsx:9]: I just tested using inspect element, it looks like the following change makes dark mode work
```suggestion
    <div className="bg-white dark:bg-dark-surface">
```

Comment (ryanchou-dev) [reply src/components/Index/ContributorsSection.tsx]: oh i just kinda removed the background

File: src/components/Index/ContributorsSection.tsx [hunk 2]
```diff
@@ -4,112 +4,96 @@ import {
   pastContentAuthors,
 } from '../../../content/authors/contributors';
 
-export default function ContributorsSection(): JSX.Element {
+export default function ContributorsSection() {
```
Comment (maggie-j-liu) [inline src/components/Index/ContributorsSection.tsx:7]: ?

File: src/components/Index/ContributorsSection.tsx [hunk 3]
```diff
@@ -4,112 +4,96 @@ import {
...
+          <div className="text-lg mx-auto">
+            <p className="text-base text-center leading-6 text-blue-600 dark:text-blue-300 font-semibold tracking-wide uppercase">
+              Contributors
+            </p>
+            <h1 className="mt-2 mb-8 text-3xl text-center leading-8 font-extrabold tracking-tight text-gray-900 dark:text-gray-100 sm:text-4xl sm:leading-10">
+              By the Community, for the Community.
+            </h1>
+            <p className="text-lg text-gray-500 dark:text-gray-400 leading-8 mb-6 mx-auto">
+              Interested in joining the USACO Guide team? We are recruiting
+              Content Authors to <br></br>
```
Comment (maggie-j-liu) [inline src/components/Index/ContributorsSection.tsx:21]: why linebreak? looks weird when window is resized :/
image

File: src/components/Index/ContributorsSection.tsx [hunk 4]
```diff
@@ -4,112 +4,96 @@ import {
...
-                  rel="noreferrer"
-                  className="text-blue-600 dark:text-blue-400"
-                >
-                  {author.name}
-                </a>
-              </li>
-            ))}
-          </ul>
+              </a>{' '}
+              If you're in the United States, you can <br></br>
```
Comment (maggie-j-liu) [inline src/components/Index/ContributorsSection.tsx:31]: same as above

File: src/components/Index/ContributorsSection.tsx [hunk 5]
```diff
@@ -4,112 +4,96 @@ import {
...
+              ))}
+            </ul>
+            <h1 className="mt-2 mb-8 text-3xl text-center leading-8 font-extrabold tracking-tight text-gray-900 dark:text-gray-100 sm:text-xl sm:leading-10">
+              Past Content Authors:
+            </h1>
+            <ul className="mx-auto grid grid-cols-2 gap-x-4 gap-y-8 sm:grid-cols-4 md:gap-x-6 lg:max-w-5xl lg:gap-x-8 lg:gap-y-12 xl:grid-cols-6">
+              {pastContentAuthors.map(author => (
+                <li key={author.name}>
+                  <div className="space-y-4">
+                    <img
```
Comment (maggie-j-liu) [inline src/components/Index/ContributorsSection.tsx:72]: is there a special component you need to use to lazyload and optimize the image? i'm not familiar w/ gatsby so idk

File: src/components/Index/ContributorsSection.tsx [hunk 6]
```diff
@@ -4,112 +4,96 @@ import {
...
+                        <a
+                          href={`https://github.com/${author.githubUsername}`}
+                          target="_blank"
+                          rel="noreferrer"
+                          className="text-blue-600 dark:text-blue-400"
+                        >
+                          <h3>{author.name}</h3>
+                        </a>{' '}
+                      </div>
+                      <p className="text-gray-500 dark:text-gray-400 list-disc list-inside text-xs space-y-2 mb-6">
```
Comment (maggie-j-liu) [inline src/components/Index/ContributorsSection.tsx:88]:
```suggestion
                      <p className="text-gray-500 dark:text-gray-400 text-xs space-y-2 mb-6">
```

File: src/components/Index/ContributorsSection.tsx [hunk 7]
```diff
@@ -4,112 +4,96 @@ import {
...
+              {pastContentAuthors.map(author => (
+                <li key={author.name}>
+                  <div className="space-y-4">
+                    <img
+                      className="mx-auto h-20 w-20 rounded-full lg:w-24 lg:h-24"
+                      src="https://i5.walmartimages.com/asr/be185d13-7b20-4595-bcdd-23bbe85f0074_1.d8c962e9921cd7f824d40d2f70adec30.jpeg?odnWidth=612&odnHeight=612&odnBg=ffffff"
+                      alt=""
+                    />
+                    <div className="space-y-2">
+                      <div className="text-xs font-medium lg:text-sm">
```
Comment (maggie-j-liu) [inline src/components/Index/ContributorsSection.tsx:78]: the text looks really small, do you think it would look better larger?

File: src/components/Index/ContributorsSection.tsx (hunk 5 above)
Comment (ryanchou-dev) [reply src/components/Index/ContributorsSection.tsx:72]: it was in the template 🤷‍♂️

Comment (maggie-j-liu) [reply src/components/Index/ContributorsSection.tsx:72]: i mean the template is for general use, but there might be something you need for gatsby? @thecodingwizard

File: src/components/Index/ContributorsSection.tsx (hunk 3 above)
Comment (ryanchou-dev) [reply src/components/Index/ContributorsSection.tsx:21]: 😳

File: src/components/Index/ContributorsSection.tsx (hunk 5 above)
Comment (thecodingwizard) [reply src/components/Index/ContributorsSection.tsx:70]: oh yes there is. use gatsby image https://www.gatsbyjs.com/plugins/gatsby-image/ (I don't remember if we use the outdated gatsby image or the newer gatsby image)

File: src/components/Index/ContributorsSection.tsx [hunk 8]
```diff
@@ -1,115 +1,133 @@
...
+              </a>{' '}
+              If you're in the United States, you can earn volunteer hours too!
+            </p>
+            <h1 className="mt-2 mb-8 text-3xl text-center leading-8 font-extrabold tracking-tight text-gray-900 dark:text-gray-100 sm:text-xl sm:leading-10">
+              Current Content Authors:
+            </h1>
+            <ul className="mx-auto grid grid-cols-2 gap-x-4 gap-y-8 sm:grid-cols-4 md:gap-x-6 lg:max-w-5xl lg:gap-x-8 lg:gap-y-12 xl:grid-cols-6">
+              {contentAuthors.map(author => (
+                <li key={author.name}>
+                  <div className="space-y-1">
```
Comment (nathangong) [inline src/components/Index/ContributorsSection.tsx:60]:
```suggestion
                  <div className="space-y-3">
```

File: src/components/Index/ContributorsSection.tsx [hunk 9]
```diff
@@ -1,115 +1,133 @@
...
+                          (data as any).allFile.edges.find(
+                            x => x.node.name === author.photo
+                          ).node.childImageSharp.gatsbyImageData
+                        }
+                        className="rounded-full overflow-hidden gatsby-image-wrapper-rounded"
+                        alt={author.name}
+                        style={{ width: '100%', height: '100%' }}
+                      />
+                    </div>
+                    <div className="space-y-2">
```
Comment (nathangong) [inline src/components/Index/ContributorsSection.tsx:73]:
```suggestion
                    <div className="space-y-1">
```

**Review by nathangong — COMMENTED** [review]
Try out this spacing: it adds more space between the image and name as well as reduces space between name and join date

**Comment by ryanchou-dev** [comment]
thanks nathan!

File: src/components/Index/ContributorsSection.tsx [hunk 10]
```diff
@@ -1,115 +1,131 @@
...
+              ))}
+            </ul>
+
+            <h1 className="mt-2 mb-8 text-3xl text-center leading-8 font-extrabold tracking-tight text-gray-900 dark:text-gray-100 sm:text-xl sm:leading-10">
+              Past Content Authors:
+            </h1>
+            <ul className="mx-auto grid grid-cols-2 gap-x-4 gap-y-8 sm:grid-cols-4 md:gap-x-6 lg:max-w-5xl lg:gap-x-8 lg:gap-y-12 xl:grid-cols-6">
+              {pastContentAuthors.map(author => (
+                <li key={author.name}>
+                  <div className="space-y-2">
```
Comment (nathangong) [inline src/components/Index/ContributorsSection.tsx:99]: This makes spacing consistent with the current content author section
```suggestion
                  <div className="space-y-3">
```

**Review by jessechoe10 — APPROVED** [review]

File: src/components/Index/ContributorsSection.tsx [hunk 11]
```diff
@@ -1,115 +1,131 @@
...
+                        <a
+                          href={`https://github.com/${author.githubUsername}`}
+                          target="_blank"
+                          rel="noreferrer"
+                          className="text-blue-600 dark:text-blue-400"
+                        >
+                          <h3>{author.name}</h3>
+                        </a>{' '}
+                      </div>
+                    </div>
```
Comment (nathangong) [inline src/components/Index/ContributorsSection.tsx:123]: you don't need the outer div because there is no "Since [date]" text, meaning you don't need to control spacing between the "name" and "since" text. if it causes an error (hopefully it doesn't), u can revert the change

```suggestion
                    <div className="text-sm font-medium lg:text-sm">
                      <a
                        href={`https://github.com/${author.githubUsername}`}
                        target="_blank"
                        rel="noreferrer"
                        className="text-blue-600 dark:text-blue-400"
                      >
                        <h3>{author.name}</h3>
                      </a>{' '}
                    </div>
```

**Review by nathangong — CHANGES_REQUESTED** [review]
Looks good! Just some super small stuff

File: src/components/Index/ContributorsSection.tsx [hunk 12]
```diff
@@ -1,115 +1,131 @@
...
+                        <a
+                          href={`https://github.com/${author.githubUsername}`}
+                          target="_blank"
+                          rel="noreferrer"
+                          className="text-blue-600 dark:text-blue-400"
+                        >
+                          <h3>{author.name}</h3>
+                        </a>{' '}
+                      </div>
+                      <p className="text-gray-500 dark:text-gray-400 list-disc list-inside text-xs space-y-2 mb-6">
```
Comment (maggie-j-liu) [inline src/components/Index/ContributorsSection.tsx:84]:
```suggestion
                      <p className="text-gray-500 dark:text-gray-400 text-xs mb-6">
```

File: src/components/Index/ContributorsSection.tsx [hunk 13]
```diff
@@ -1,115 +1,131 @@
...
+                    <div className="space-y-1">
+                      <div className="text-sm font-medium lg:text-sm">
+                        <a
+                          href={`https://github.com/${author.githubUsername}`}
+                          target="_blank"
+                          rel="noreferrer"
+                          className="text-blue-600 dark:text-blue-400"
+                        >
+                          <h3>{author.name}</h3>
+                        </a>{' '}
```
Comment (maggie-j-liu) [inline src/components/Index/ContributorsSection.tsx:82]:
```suggestion
                        </a>
```

File: src/components/Index/ContributorsSection.tsx [hunk 14]
```diff
@@ -1,115 +1,131 @@
...
+                    <div className="space-y-2">
+                      <div className="text-sm font-medium lg:text-sm">
+                        <a
+                          href={`https://github.com/${author.githubUsername}`}
+                          target="_blank"
+                          rel="noreferrer"
+                          className="text-blue-600 dark:text-blue-400"
+                        >
+                          <h3>{author.name}</h3>
+                        </a>{' '}
```
Comment (nathangong) [inline src/components/Index/ContributorsSection.tsx:121]:
```suggestion
                        </a>
```

File: content/authors/contributors.ts [hunk 15]
```diff
@@ -2,73 +2,87 @@ export type Contributor = {
...
   sinceDate: string;
+  photo: string;
 };
 
 export const contentAuthors: Contributor[] = [
   {
     name: 'Jeffrey Meng',
     githubUsername: 'jeffreymeng',
     sinceDate: 'Oct 2020',
+    photo: 'jeffery',
```
Comment (maggie-j-liu) [inline content/authors/contributors.ts:13]: misspelled

Comment (ryanchou-dev) [reply content/authors/contributors.ts:13]: oops

File: content/authors/contributors.ts [hunk 16]
```diff
@@ -2,73 +2,87 @@ export type Contributor = {
...
     name: 'Kevin Sheng',
     githubUsername: 'sanspapyrus683',
     sinceDate: 'Apr 2021',
+    photo: 'kevin',
   },
   {
     name: 'Ananth Kashyap',
     githubUsername: 'AAkashyap1',
     sinceDate: 'Apr 2021',
+    photo: 'anath',
```
Comment (dongliu0426) [inline content/authors/contributors.ts:37]: typo

Comment (ryanchou-dev) [reply content/authors/contributors.ts]: fixed :) thanks!
